File: py-server-app/src/person.py
class Person:
    def __init__(self, name: str, weight: float, hobbies: list):
        self.name = name
        if isinstance(weight, str) and weight.replace('.', '', 1).isdigit():
            self.weight = float(weight)
        elif isinstance(weight, (int, float)):
            self.weight = float(weight)
        else:
            self.weight = weight
        self.hobbies = hobbies

    def greet(self) -> str:
        return f"Hello, my name is {self.name} and I weigh {self.weight} kg. and my hobbies are {', '.join(self.hobbies)}"

    def __repr__(self) -> str:
        hobbies_str = ', '.join(f"'{hobby}'" for hobby in self.hobbies)
        return f"Person {{\n  name: '{self.name}',\n  weight: {self.weight},\n  hobbies: [ {hobbies_str} ]\n}}"
    

File: js-server-app/src/person.js
class Person {
    constructor(name, weight, hobbies) {
        this.name = name;
        if (typeof weight === 'string' && !isNaN(value)) {
            this.weight = parseFloat(weight);
        }
        else if (typeof weight === 'number') {
            this.weight = parseFloat(weight);
        } else {
            throw new Error('Weight must be a number or a string that can be converted to a number');
        }
        this.hobbies = hobbies;
    }
    greet() {
        return `Hello, my name is ${this.name} and I weigh ${this.weight} kg. and my hobbies are ${this.hobbies.join(', ')}`;
    }
}

export default Person;